src/components/text-editor/text-editor.component.tsx

```diff
@@ -62,41 +98,61 @@ const TextEditor = React.forwardRef(
       previews,
       onLinkAdded,
       ...rest
-    },
+    }: TextEditorProps,
     ref
   ) => {
     const [isFocused, setIsFocused] = useState(false);
-    const [inlines, setInlines] = useState([]);
-    const [activeInlines, setActiveInlines] = useState({});
+    const [inlines, setInlines] = useState<InlineStyleType[]>([]);
+    const [activeInlines, setActiveInlines] = useState<{
+      BOLD?: boolean;
+      ITALIC?: boolean;
+    }>({});
     const [focusToolbar, setFocusToolbar] = useState(false);
 
-    const editorRef = useRef();
+    const editorRef = useRef<Editor>(null);
     const editor = ref || editorRef;
     const contentLength = getContent(value).getPlainText("").length;
     const moveCursor = useRef(contentLength > 0);
-    const lastKeyPressed = useRef();
+    const lastKeyPressed = useRef<null | string>();
     const labelId = useRef(`text-editor-label-${createGuid()}`);
 
-    const keyBindingFn = (ev) => {
+    const invalidRowProp = rows && (typeof rows !== "number" || rows < 2);
+    invariant(
+      !invalidRowProp,
+      `Prop rows must be a number value greater than 2 to override the min-height of the \`${TextEditor.displayName}\``
```

Comment:
The message should be something like:
`Prop rows must be a number value that is 2 or greater than 2 to override the min-height...`

I put this as an invariant as I thought that was what agreed when we first started doing the TS refactors. This might have changed whilst I was off sick for a while though and I've just presumed we still do invariants? Happy to change to a `console.warn` as you say, it'll stop the potential crash of apps.

Reply:
I'm not sure tbh, might be worth asking more widely. We definitely do still use invariants, I'm just nervous of introducing a new one when we've had consumers complain about this in the past - but we've probably added them in TS refactors for similar cases to this one, so I dunno really.